crates/uv-cli/src/lib.rs

```diff
@@ -1999,8 +2000,8 @@ pub struct VenvArgs {
     ///
     /// Defaults to `clone` (also known as Copy-on-Write) on macOS, and `hardlink` on Linux and
     /// Windows.
-    #[arg(long, value_enum, env = "UV_LINK_MODE")]
-    pub link_mode: Option<install_wheel_rs::linker::LinkMode>,
+    #[arg(long, value_enum, env = "UV_LINK_MODE", default_value_t=LinkMode::default())]
+    pub link_mode: LinkMode,
```

Comment:
I think we can't do this because we need to know if the value is provided on the CLI _at all_.

Reply:
Ok, I think the best approach is to use the `value_source` to determine where the value is being set. Then, if the source is not from the command line, we can simply reset it to `None`.

Reply:
Another simpler but trickier approach would be to leverage non-printable characters.

Consider the following:
```rust
#[derive(Clone, Debug)]
pub enum MaybeGiven<T> {
    Default(T),
    Given(T),
}

impl<T: Default> Default for MaybeGiven<T> {
    fn default() -> Self {
        Self::Default(T::default())
    }
}

impl<T> MaybeGiven<T> {
    pub fn into_given(self) -> Option<T> {
        match self {
            MaybeGiven::Default(_) => None,
            MaybeGiven::Given(x) => Some(x),
        }
    }

    pub fn into_inner(self) -> T {
        match self {
            MaybeGiven::Default(x) => x,
            MaybeGiven::Given(x) => x,
        }
    }
}

impl<T> AsRef<T> for MaybeGiven<T> {
    fn as_ref(&self) -> &T {
        match self {
            MaybeGiven::Default(x) => x,
            MaybeGiven::Given(x) => x,
        }
    }
}

impl<T: Display> Display for MaybeGiven<T> {
    fn fmt(&self, f: &mut std::fmt::Formatter<'_>) -> std::fmt::Result {
        if matches!(self, MaybeGiven::Default(_)) {
            f.write_str("\0")?;
            self.as_ref().fmt(f)?;
            f.write_str("\0")
        } else {
            self.as_ref().fmt(f)
        }
    }
}

impl<T: FromStr> FromStr for MaybeGiven<T> {
    type Err = T::Err;

    fn from_str(s: &str) -> Result<Self, Self::Err> {
        let trimmed = s.trim_matches('\0');
        if trimmed != s {
            return Ok(Self::Default(T::from_str(trimmed)?));
        }
        Ok(Self::Given(T::from_str(s)?))
    }
}
```

It checks if the string is surrounded by '\0' to determine whether it is from the default value or given by the user. This is a more hacky but simpler implementation without modifying the existing codebase too much, IMO.

If this is considered too hacky, I could implement it by checking the `value_source`.